File: ruby/hamming/hamming.rb
class Hamming
  Hamming::VERSION = 1

  def self.compute(string, string2)
    raise ArgumentError if string.length != string2.length

    array = string.chars.zip(string2.chars)

    array.count do |pair1, pair2|
      pair1 != pair2
    end
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(dna1, dna2) {
  var count = 0;
  if (dna1.length != dna2.length) {
    throw new Error('DNA strands must be of equal length.')
  }
  for (i = 0; i < dna1.length; i++) {
    if (dna1[i] != dna2[i]) {
      // console.log(dna1[i], dna2[i])
      count += 1;
    }
  }
  return count
};

module.exports = Hamming;
